encrypted-media-respec.html

```diff
@@ -605,8 +605,112 @@
 
           <section>
             <h5>Get Supported Configuration</h5>
-            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>null</code> as appropriate.</p>
-            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.</p>
+            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>NotSupported</code> as appropriate.</p>
+            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.
+            </p>
+            <div class="note">
+              <p>
+              For certain configurations, it may be required to obtain user consent or inform the user. User Agents have some flexibility to determine
+              whether consent is required for a specific configuration and whether such consent may also apply to other configurations. For example,
+              consent to one configuration may also imply consent for less powerful, more restricted configurations. Equally, a denial of consent for
+              one configuration may imply denial of consent for more powerful, less restricted configurations.
+              </p>
+              <p>
+              Supported configurations, including supported audio and video codecs, may depend on availability of optional capabilities such as
+              <a def-id="distinctive-identifier"></a>s and persistent state. The following algorithm iteratively tries configurations with different
+              requirement values for these optional configuration elements until consent is achieved.
+              </p>
+              <p>
+              User Agents should persist consent responses at least for the duration of the <a def-id="requestMediaKeySystemAccess"></a>
+              algorithm in order to avoid repeated requests to the user for similar configurations.
+              </p>
+            </div>
+            <ol>
+              <li>
+                <p>Let <var>requirements</var> be the list [ <code>distinctiveIdentifier</code>, <code>persistentState</code> ].</p>
```

Comment:
>I don't believe we can handle this with the existing text.

To make sure I understand, is this because DI is set to `not-allowed` based on the _restrictions_ list?


<br/>Thanks for listing the bullets. My idea as slightly different (see the paragraph above related to "handl[ing] lack of consent for other things"), and I think it might address this problem. Thus, the changes will be:
- remove the construction of "all possible combinations"
- change the loop to "repeat until done"**, which is probably until the result is not `ConsentDenied`**
- Get Supported Configurations and Consent will return a _restrictions_ list with the `ConsentDenied` return value. The _restrictions_ list is a list of those **properties ~~elements of the _modified configuration_ of type `MediaKeyRequirement` which had value `optional` and~~** for which consent has been denied. **The contents of _restrictions_ is implementation-specific, but it likely has at least one property related to DI.**
- **~~If _restrictions_ is non-empty, update _modified candidate configuration_ to set all the members listed in _restrictions_ to `not-allowed` and continue the loop~~ The _restrictions_ list is passed to the method, which is updated to consider it wherever (support in combination with) _accumulated configuration_ is considered. The value of the "optional" values is not changed, but it is likely that "required" is no longer a valid value when _restrictions_ is considered.**
- **~~If _restrictions_ is empty, it means none of the `MediaKeyRequirement` elements were optional and consent was denied anyway, so we return `NotSupported`.~~ We continue to return `NotSupported` whenever we reach a point where there are no supported configurations, including due to the application of _restrictions_. (_restrictions_ is only returned/considered when `ConsentDenied` is returned.) This allows non-consent-related failures to work as before while also addressing the loop termination.**